client.go

```diff
@@ -233,30 +256,36 @@ func WithDomainSearchLimit(limit int) Option {
 // Name is derived from root if possible.
 // Root is defaulted to the current working directory.
 func (c *Client) Create(language, name, root string) (err error) {
+	c.progressListener.SetTotal(5)
```

Comment:
I agree.  This is intended to be used as a library primarily, a CLI secondarily, so relying on the os.Exit in the CLI on errors works for now, but you are right this should be a defer statement to ensure it gets run.

Reply:
Also note that regarding the defer statement, a "defer Stop()" was used rather than deferring the Complete statement, because otherwise if the function returns mid-way through execution, the "Complete" message is printed.  This is inaccurate (if 3/5 steps are completed, the completion event did not occur so the text "5/5 Create Complete" would be a bit msleading!  The deferred Stop() simply ensures the print write is cancelled.